src/ConnectedMode/UI/ProjectSelection/ProjectSelectionViewModel.cs

```diff
@@ -58,15 +65,34 @@ public ServerProject SelectedProject
     private string projectSearchTerm;
     private ServerProject selectedProject;
 
-    public ProjectSelectionViewModel(ConnectionInfo connectionInfo)
+    public async Task InitializeProjectWithProgressAsync()
+    {
+        var initializeProjectsParams = new TaskToPerformParams<AdapterResponseWithData<List<ServerProject>>>(
+            AdapterGetAllProjectsAsync,
+            UiResources.LoadingProjectsProgressText,
+            UiResources.LoadingProjectsFailedText)
+        {
+            AfterSuccess = InitProjects
+        };
+        await ProgressReporterViewModel.ExecuteTaskWithProgressAsync(initializeProjectsParams);
+    }
+
+    internal async Task<AdapterResponseWithData<List<ServerProject>>> AdapterGetAllProjectsAsync()
     {
-        ConnectionInfo = connectionInfo;
+        var serverConnectionCredentials = connectedModeServices.ServerConnectionsRepositoryAdapter.TryGet(ConnectionInfo.Id, out var serverConnection);
+        if (!serverConnectionCredentials)
+        {
+            connectedModeServices.Logger.WriteLine(UiResources.LoadingProjectsFailedTextForNotFoundServerConnection);
+            return new AdapterResponseWithData<List<ServerProject>>(false, null);
+        }
+
+        return await connectedModeServices.SlCoreConnectionAdapter.GetAllProjectsAsync(ConnectionInfo, serverConnection.Credentials);
     }
 
-    public void InitProjects(List<ServerProject> projects)
+    internal void InitProjects(AdapterResponseWithData<List<ServerProject>> responseData)
     {
         ProjectResults.Clear();
-        projects.ForEach(x => ProjectResults.Add(x));
+        responseData.ResponseData.ForEach(x => ProjectResults.Add(x));
```

Comment:
Can we sort the results?

Especially because we don't support searching at the moment it's kinda hard for the user to find their projects.

Reply:
Makes total sense. Good idea!